cli/threadexecutor.cpp

```diff
@@ -486,32 +454,32 @@ unsigned int __stdcall ThreadExecutor::threadProc(void *args)
         } else { // file settings..
             const ImportProject::FileSettings &fs = *itFileSettings;
             ++itFileSettings;
-            LeaveCriticalSection(&threadExecutor->mFileSync);
+            threadExecutor->mFileSync.unlock();
             result += fileChecker.check(fs);
             if (threadExecutor->mSettings.clangTidy)
                 fileChecker.analyseClangTidy(fs);
         }
 
-        EnterCriticalSection(&threadExecutor->mFileSync);
+        threadExecutor->mFileSync.lock();
 
         threadExecutor->mProcessedSize += fileSize;
         threadExecutor->mProcessedFiles++;
         if (!threadExecutor->mSettings.quiet) {
-            EnterCriticalSection(&threadExecutor->mReportSync);
+            threadExecutor->mReportSync.lock();
             CppCheckExecutor::reportStatus(threadExecutor->mProcessedFiles, threadExecutor->mTotalFiles, threadExecutor->mProcessedSize, threadExecutor->mTotalFileSize);
-            LeaveCriticalSection(&threadExecutor->mReportSync);
+            threadExecutor->mReportSync.unlock();
         }
     }
     return result;
 }
 
 void ThreadExecutor::reportOut(const std::string &outmsg, Color c)
 {
-    EnterCriticalSection(&mReportSync);
```

Comment:
can we use some scoped locks in some of these places?

Reply:
Sure. I'm out of office right now, so I'll get to it next week.

Reply:
ok no problem thanks for looking at this!